tensorboard/components/tf_backend/requestManager.ts

```diff
@@ -207,14 +210,16 @@ export class RequestManager {
   /* Actually get promise from url using XMLHttpRequest */
   protected _promiseFromUrl(url: string, requestOptions: RequestOptions) {
     return new Promise((resolve, reject) => {
-      const req = buildXMLHttpRequest(
-        requestOptions.methodType,
-        url,
-        requestOptions.withCredentials,
-        requestOptions.contentType);
+      const req = buildXMLHttpRequest(url, requestOptions);
       req.onload = function() {
         if (req.status === 200) {
-          resolve(JSON.parse(req.responseText));
+          // Edge and IE do not support "responseType == 'json'".
+          // Fall back to manual parsing.
+          if (req.responseType == 'json' && typeof req.response == 'string') {
```

Comment:
I’m not sure that this test actually works in IE. On my machine:


```js
var xhr = new XMLHttpRequest();
xhr.open("GET", "http://example.com", true);
function report() {
  console.log("xhr.responseType: %s", JSON.stringify(xhr.responseType));
}

report();  // ""
xhr.responseType = "json";
report();  // ""
xhr.responseType = "blob";
report();  // "blob"
xhr.responseType = "json";
report();  // "blob"
```

(This is IE11 on Windows 10.)

Reply:
Good call in a sense that I should have tried it on my Windows machine instead of assuming things based on MDN doc.

In any case (I will come back with my own experiments), this seems to be "correct". In case response CANNOT be parsed as the desired type, the `response` is an empty string. Just making sure, is your code snippet correct?

Reply:
> In case response CANNOT be parsed as the desired type, the `response`
> is an empty string.

As far as I can tell:

 1. If the **response** from the server cannot be parsed as the desired
    type, then the `response` property of the XHR is `null` (at least
    when `responseType` is `"json"` on Chrome M73 on Linux).

 2. If one tries to set the `responseType` **property** is to an invalid
    response type (like `"json"` in IE11), then this property is not
    changed from its previous value. In particular, in IE11, it is not
    possible to create an XHR such that `req.responseType === "json"`.

The first point isn’t really what I’m interested in. I’m concerned that
the check `req.responseType == 'json'` will always evaluate to `false`
in IE, which means that control will never flow into the very block
that’s supposed to handle IE compatibility.

It’s totally plausible that I’m wrong about some or all of this; the XHR
spec and implementations are certainly not my expertise.

> Just making sure, is your code snippet correct?

![Screenshot of code snippet in IE11](https://user-images.githubusercontent.com/4317806/56444264-da6d3d80-62ac-11e9-8869-12729f0c80c0.png)